### ISSUE TITLE
partial payments error

### ISSUE THREAD

**Issue by samuelbirch** [craftcms/commerce#2501]
if a partial payment value is set to 0 then the payment amount uses the outstanding balance instead of 0

```
public function getPaymentAmount(): float
    {
        $outstandingBalanceInPaymentCurrency = Plugin::getInstance()->getPaymentCurrencies()->convertCurrency($this->getOutstandingBalance(), $this->currency, $this->paymentCurrency);

        if ($this->_paymentAmount && $this->_paymentAmount >= 0 && $this->_paymentAmount <= $outstandingBalanceInPaymentCurrency) {
            return $this->_paymentAmount;
        }

        return $outstandingBalanceInPaymentCurrency;
    }
```

should be:

```
public function getPaymentAmount(): float
    {
        $outstandingBalanceInPaymentCurrency = Plugin::getInstance()->getPaymentCurrencies()->convertCurrency($this->getOutstandingBalance(), $this->currency, $this->paymentCurrency);

        if (isset($this->_paymentAmount) && $this->_paymentAmount >= 0 && $this->_paymentAmount <= $outstandingBalanceInPaymentCurrency) {
            return $this->_paymentAmount;
        }

        return $outstandingBalanceInPaymentCurrency;
    }
```

**Comment by lukeholder** [comment]
Thanks, this has been fixed for the next release.

To get the fix early, change your `craftcms/commerce` requirement in composer.json to:

```json
"require": {
  "craftcms/commerce": "dev-develop#d73a86f6f2ade5725a571ad68269691da7897755 as 3.4.1",
  "...": "..."
}
```

Then run `composer update`.

**Comment by lukeholder** [comment]
This is now fixed in 3.4.2 - you can update your compose back to the normal version track. Thanks.
